soc/renesas/smartbond/da1469x/soc.c

```diff
@@ -36,9 +37,27 @@ static uint32_t z_renesas_cache_configured;
 
 void sys_arch_reboot(int type)
 {
-	ARG_UNUSED(type);
-
-	NVIC_SystemReset();
+	if (type == SYS_REBOOT_WARM) {
+		NVIC_SystemReset();
+	} else if (type == SYS_REBOOT_COLD) {
+		if ((SYS_WDOG->WATCHDOG_REG & SYS_WDOG_WATCHDOG_REG_WDOG_VAL_NEG_Msk) == 0) {
+			/* Cannot write WATCHDOG_REG while WRITE_BUSY */
+			while ((SYS_WDOG->WATCHDOG_REG &
+				SYS_WDOG_WATCHDOG_CTRL_REG_WRITE_BUSY_Msk) != 0) {
+
+			}
+			/* Write WATCHDOG_REG */
+			SYS_WDOG->WATCHDOG_REG = BIT(SYS_WDOG_WATCHDOG_REG_WDOG_VAL_Pos);
+
+			GPREG->RESET_FREEZE_REG = GPREG_SET_FREEZE_REG_FRZ_SYS_WDOG_Msk;
```

Comment:
Freezing/resuming the watchdog counter requires that `WATCHDOG_CTRL_REG[WDOG_FREEZE_EN]` is first written to one.

Reply:
The goal here is to let the watchdog running and cause a reset.
If `WATCHDOG_CTRL_REG[WDOG_FREEZE_EN] == 0` then the watchdog is running and we are good.
If `WATCHDOG_CTRL_REG[WDOG_FREEZE_EN] == 1` we make sure we unfreeze it.
The same stands for `WATCHDOG_CTRL_REG[NMI_RST]`.